tests/functional/utils/chart_certification.py

```diff
@@ -0,0 +1,752 @@
+                        logging.warning(
+                            f"Error parsing OWNERS of {chart_directory}: {err}")
+            else:
+                owners_table[chart_directory] = self.secrets.notify_id
+
+    def push_chart(self, chart_directory, chart_name, chart_version, vendor_name, vendor_type, pr_branch):
+        # Push chart files to test_repo:pr_branch
+        self.temp_repo.git.add(f'{chart_directory}/{chart_version}')
+        self.temp_repo.git.commit(
+            '-m', f"Add {vendor_type} {vendor_name} {chart_name} {chart_version} chart files")
+        self.temp_repo.git.push(f'https://x-access-token:{self.secrets.bot_token}@github.com/{self.secrets.test_repo}',
+                    f'HEAD:refs/heads/{pr_branch}', '-f')
+
+    def check_single_chart_result(self, vendor_type, vendor_name, chart_name, chart_version, pr_number, owners_table):
+        base_branch = f'{self.secrets.software_name}-{self.secrets.software_version}-{self.secrets.pr_base_branch}-{vendor_type}-{vendor_name}-{chart_name}-{chart_version}'
+
+        # Check PRs are merged
+        chart = f'{vendor_type} {vendor_name} {chart_name} {chart_version}'
+        run_id = get_run_id(self.secrets, pr_number)
+        conclusion = get_run_result(self.secrets, run_id)
+
+        # Send notification to owner through GitHub issues
+        if not self.secrets.dry_run:
+            r = github_api(
+                'get', f'repos/{self.secrets.test_repo}/actions/runs/{run_id}', self.secrets.bot_token)
+            run = r.json()
+            run_html_url = run['html_url']
+            chart_directory = f'charts/{vendor_type}/{vendor_name}/{chart_name}'
+            chart_owners = owners_table[chart_directory]
+            pass_verification = conclusion == 'success'
+            os.environ['GITHUB_ORGANIZATION'] = PROD_REPO.split('/')[0]
+            os.environ['GITHUB_REPO'] = PROD_REPO.split('/')[1]
+            os.environ['GITHUB_AUTH_TOKEN'] = self.secrets.bot_token
+            logging.info(
+                f"Send notification to '{chart_owners}' about verification result of '{chart}'")
+            create_verification_issue(chart_name, chart_owners, run_html_url, self.secrets.software_name,
+                                    self.secrets.software_version, pass_verification, self.secrets.bot_token)
+
+        if conclusion == 'success':
+            logging.info(f"Workflow run for {chart} was 'success'")
+        else:
+            logging.warning(
+                f"Workflow for the submitted PR did not success, run id: {run_id}, chart: {chart}")
+            return
+
+        r = github_api(
+            'get', f'repos/{self.secrets.test_repo}/pulls/{pr_number}/merge', self.secrets.bot_token)
+        if r.status_code == 204:
+            logging.info(f"PR for {chart} merged sucessfully")
+        else:
+            logging.warning(
+                f"Workflow for submitted PR success but PR not merged, chart: {chart}")
+            return
+
```

Comment:
can we reuse the check_pull_request_result() function as defined in "ChartCertificationE2ETestSingle" data class and make check_pull_request_result() as part of "ChartCertificationE2ETest" data class ? Above portion of code looks similar.

Reply:
Lifted the common methods into the parent class. Updated